### PR TITLE
Add support for nested resources

### PR DESCRIPTION
**Pull request by pmen**
At time of resource definition, support defining a mapping of `message field name` => `Protip::Resource class` that will be used to wrap those message fields in the defined Resource. 

The nested resources will behave identically to resources instantiated manually with the same message.

### CODE REVIEW

File: lib/protip/wrapper.rb [hunk 1]
```diff
@@ -201,6 +208,8 @@ def to_protobuf_value(field, value)
           value
         elsif converter.convertible?(field.subtype.msgclass)
           converter.to_message value, field.subtype.msgclass
+        elsif nested_resources.has_key?(field.name.to_sym)
+          value.message
```
Comment (tubergen) [inline lib/protip/wrapper.rb]: this assumes that a resource is passed in as `value`, but currently, lots of our associations pass in messages

i guess it will be fine, since those "associations" won't be declared as associations yet. i guess when we declare an association, we need to carefully change that model's `as_proto` ? (i'm not sure we have very good integration test coverage of protip stuff, so we'll have to be very careful)

File: test/unit/protip/wrapper_test.rb [hunk 2]
```diff
@@ -60,6 +60,31 @@ module Protip::WrapperTest # namespace for internal constants
...
+    # Stubbed API client
+    let :client do
+      mock.responds_like_instance_of(Class.new { include Protip::Client })
+    end
+
+    let :converter do
+      Class.new do
+        include Protip::Converter
+      end.new
+    end
```
Comment (tubergen) [inline test/unit/protip/wrapper_test.rb]: don't we have this same code above?

File: test/unit/protip/wrapper_test.rb [hunk 3]
```diff
@@ -346,6 +371,13 @@ module Protip::WrapperTest # namespace for internal constants
     end
 
     describe '#get' do
+      before do
+        resource_class.class_exec(converter, inner_message_class) do |converter, message|
```
Comment (tubergen) [inline test/unit/protip/wrapper_test.rb]: can you explain what this is doing?

File: test/unit/protip/wrapper_test.rb [hunk 4]
```diff
@@ -414,6 +462,20 @@ module Protip::WrapperTest # namespace for internal constants
...
+        klass = resource_class
+        new_inner_message = inner_message_class.new(value: 50)
+
+        resource = klass.new new_inner_message
+        wrapper = Protip::Wrapper.new(message, Protip::StandardConverter.new, {inner: klass})
+
+        resource.expects(:message).once.returns(new_inner_message)
+        wrapper.inner = resource
+
+        assert_equal new_inner_message, wrapper.message.inner
```
Comment (tubergen) [inline test/unit/protip/wrapper_test.rb]: overall, it's hard to follow this test IMO. one thing that might help would be to add an error message as the third arg describing the failure case.

File: test/unit/protip/wrapper_test.rb (hunk 2 above)
Comment (pmen) [inline test/unit/protip/wrapper_test.rb]: in a different test case, yes.

File: test/unit/protip/wrapper_test.rb (hunk 3 above)
Comment (pmen) [inline test/unit/protip/wrapper_test.rb]: It's defining a resource with no actions, a message of inner_message_class, and using the converter defined above
